pkg/utils/utils.go

```diff
@@ -234,3 +236,25 @@ func SetAvailableResourcesForApi(groupVersion string, resources *metav1.APIResou
 
 	gvResourcesCache[groupVersion] = resources
 }
+
+func GetEnvVarValue(envVars []v1.EnvVar, key string) (string, bool) {
+	for _, envVar := range envVars {
+		if envVar.Name == key {
+			return envVar.Value, true // if key exist, return value, true
+		}
+	}
+	return "", false // if key does not exist, return "", false
+}
+
+func ConvertStringToInt32(number int, defaultValue int) (int32, error) {
+	// Check if the number exceeds the int32 limits
+	if number > math.MaxInt32 || number < math.MinInt32 {
+		// Also check the defaultValue to ensure it's safe to convert
```

Comment:
Any tests that cover these new functions?

Reply:
if the default value is hardcoded, why check it?

Reply:
added

Reply:
if we adopt the use of the `intstr` this method could be removed, wdyt?
IIRC all fields like port, targetPort, etc from a K8s resource use intstr as type.

Reply:
yes, I removed ConvertStringToInt32